app/templates/racer/index.html

```diff
@@ -18,124 +23,108 @@
+      {% for race_class in race_classes.values()|sort(attribute="name") %}
+      <label class="btn btn-primary">
+        <input id="class-checkbox-{{race_class.id}}" type="checkbox"
+               autocomplete="off" value="{{race_class.id}}">{{race_class.name}}
+      </label>
+      {% endfor %}
+    </div>
+  </div>
+</fieldset>
 
-<div class="table-responsive">
-<table class="table table-striped table-bordered">
+<div id="racers-table-div" class="table-responsive">
+<table id="racers-table" class="table table-striped table-bordered">
 <thead>
 <tr>
-    <th>Name</th>
-    <th>Team</th>
+    <th data-field="name">Name</th>
+    <th data-field="current_team_name">Team</th>
 </tr>
 </thead>
 <tbody>
-{% for racer in racers %}
-<tr>
-<td><a href="{{url_for('racer.details',id=racer.id)}}">{{racer.name}}</a>
-{%if racer.strava_id%}
-<style>
-  .strava-badge- { display: inline-block; height: 24px;}
-  .strava-badge- img { visibility: hidden; height: 24px; }
-  .strava-badge-:hover { background-position: 0 -37px; }
-  .strava-badge-follow { height: 24px; width: 24px; background: url(//badges.strava.com/echelon-sprite-24.png) no-repeat 0 0; }
-</style>
-<a href="http://strava.com/athletes/{{racer.strava_id}}/badge" class="strava-badge- strava-badge-follow" target="_blank"><img src="//badges.strava.com/echelon-sprite-24.png" alt="Strava" /></a>
-{%endif%}
-</td> 
-{%if racer.current_team %}
-<td><a href="{{url_for('team.details',id=racer.current_team.id)}}">{{racer.current_team.name}}</a></td>
-{%else%}
-<td></td>
-{%endif%}
-</tr>
-{% endfor %}
-</tbody>
 </table>
 </div>
 
 {% endblock %}
+
+{% block scripts %}
+{{super()}}
+<script src="{{url_for('static', filename='bootstrap-slider.min.js')}}"></script>
+<script>
+function updateDocument() {
+  var isNoRaces = $('#ckb-no-races').prop('checked');
+  $('#season-slider').slider(isNoRaces ? 'disable' : 'enable');
+  $('#fs-class-checkboxes').prop('disabled', isNoRaces);
+
+  var seasonRange = $('#season-slider').slider('getValue').join();
+  var classFilters = $('#class-checkboxes input')
+    .filter(function() { return this.checked; })
+    .map(function() { return this.value; }).get().join();
+
+  args = isNoRaces ? {} :
+    {'filter_seasons': seasonRange,
+     'filter_race_class_ids': classFilters};
+  $.get(
+    "/racer/json/filter", args,
+    function(data) {
+      var rows = ''
+      $.each(JSON.parse(data), function(i, racer) {
+        var row = '<tr>'
+
+        row += '<td>'
+        row += '<a href="/racer/' + racer.id + '">' + racer.name + '</a>'
+        if (racer.strava_id) {
+          row += '<style>'
+          row += '.strava-badge- { display: inline-block; height: 24px;} '
+          row += '.strava-badge- img { visibility: hidden; height: 24px; }'
+          row += '.strava-badge-:hover { background-position: 0 -37px; }'
+          row += '.strava-badge-follow { height: 24px; width: 24px; background: url(//badges.strava.com/echelon-sprite-24.png) no-repeat 0 0; }'
```

Comment:
But because you've made some pretty significant changes to `app/templates/racer/index.html`, will this merge cleanly?